source/fab/tools/ar.py

```diff
@@ -0,0 +1,46 @@
+##############################################################################
+# (c) Crown copyright Met Office. All rights reserved.
+# For further details please refer to the file COPYRIGHT
+# which you should have received as part of this distribution
+##############################################################################
+
+"""This file contains the Ar class for archiving files.
+"""
+
+from pathlib import Path
+from typing import List, Union
+
+from fab.tools.categories import Categories
+from fab.tools.tool import Tool
+
+
+class Ar(Tool):
+    '''This is the base class for `ar`.
+    '''
+
+    def __init__(self):
+        super().__init__("ar", "ar", Categories.AR)
+
+    def check_available(self):
+        '''
+        :returns: whether `ar` is available or not. We do this by
+            requesting the ar version.
+        '''
+        try:
+            self.run("--version")
```

Comment:
Hmm - not sure which one is better. Could we have an executable in the path that is not executable? While certainly not applicable for any useful example here, what if we have a dynamic binary without RPATH and incorrect LD_LIBRARY_PATH? That would probably still report under `which`, but not be executable?
``` 
~/fab-workspace/skeleton-mpif90$ which ./skeleton
./skeleton
:~/fab-workspace/skeleton-mpif90$ ./skeleton 
./skeleton: error while loading shared libraries: libyaxt.so.1: cannot open shared object file: No such file or directory
```

But I am happy to change, I don't think it will make a difference in reality.